modules/ui/sections/raw_membership_editor.js

```diff
@@ -205,15 +206,64 @@ export function uiSectionRawMembershipEditor(context) {
     }
 
 
-    function downloadMembers(d3_event, d) {
+    async function downloadMembers(d3_event, d) {
         d3_event.preventDefault();
         const button = d3_select(this);
 
-        // display the loading indicator
-        button.classed('loading', true);
-        context.loadEntity(d.relation.id, function() {
+        if (d.relation.members.length > 250) {
```

Comment:
> If the relation has more than ~300 members, clicking the button will crash iD.

In my experience, 300 members isn’t that extreme and definitely wouldn’t risk a crash in general. [A mid-sized U.S. state](https://www.openstreetmap.org/relation/162061) or [a statewide bike trail](https://www.openstreetmap.org/relation/118953) would easily surpass this threshold. A confirmation dialog wouldn’t hurt, but the wording doesn’t need to be so certain and alarming. If this prevents the user from downloading the full relation, that could be unfortunate. Is there any way we could offer a third option to download the full relation at your own risk?

[Mainland Australia](https://www.openstreetmap.org/relation/4095122)[^boundary] has 3,583 members, which is a lot, but downloading it works for me without crashing in Firefox 134. It’s just very slow. I regularly work with [much larger relations](https://www.openhistoricalmap.org/relation/2798718) in OpenHistoricalMap’s fork of iD that don’t incur any noticeable performance penalty. The performance issue is probably a function of the total number of nodes rather than the number of members, and maybe other factors too. For example, [this multipolygon](https://www.openhistoricalmap.org/relation/2746142), which represents some marshes on the outskirts of San José, California, has only three members, but drawing an area that intersects it slows down iD even more than mainland Australia for me.

[^boundary]: Why the heck is there a `boundary` relation for a continent?

Reply:
yeah, I wonder if this is the total wrong approach. #10316 is the root of the problem, which this doesn't even't solve. This is more of a hack so I can edit my local beach 😅

I'm starting to think that we need a different popup which interrupts the split operation, and offers to download the prev+next members. And then we leave this download button alone.